File: Lecture06/StaticTest2.py
class Element:
  _numElements = 0
  
  def __init__(self):
    Element._numElements += 1
    self.id = Element._numElements

  @classmethod
  def getNumElements(cls):
    return cls._numElements  

if __name__=="__main__":
  # x
  y = Element()
  z = Element()
  print("Number of Elements created:",Element.getNumElements())
  print("Element Id of y:",y.id)

File: Lecture06/StaticTest2.java
class Element {
  private static int numElements = 0;
  public int id;
  
  Element() {
    id = ++numElements;
  }

  public static int getNumElements() {
    return numElements;
  }
}

public class StaticTest2 {
  public static void main(String[] args) {
    Element x;
    Element y = new Element();
    Element z = new Element();
    System.out.println("Number of Elements created: " + Element.getNumElements());
    System.out.println("Element Id of y: " + y.id);
  }
}
